bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/util/StringUtils.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.io.neeo.internal.util;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * The {@link StringUtils} class defines some static string utility methods
+ *
+ * @author Leo Siepel - Initial contribution
+ */
+@NonNullByDefault
+public class StringUtils {
+
+    public static String[] split(@Nullable String input, String delimiter) {
```

Comment:
Looking at the usages, I don't think "support" for null is needed?

Reply:
dropped null support, i just liked it as it is more generic.

Reply:
Well, a method accepting `null` as parameter and returning `null` from that is indeed very generic. 😉 Too generic to be useful, IMHO.

In general, for a public method, if such usage doesn't make sense, it would be preferable to throw [IllegalArgumentException](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/IllegalArgumentException.html) to fail fast, and avoid calling it with a `null` parameter, i.e. not make it the responsibility of external methods to handle such error/corner case condition.

Internally we can often use null annotations to avoid such boiler-plate code and instead get help from static code analysis.

Reply:
Actually it was returning an empty array, not null. It is how many utility methods are graceful handling null cases. Your suggestion is also a very good pattern, I’ll try to keep it in mind. But anyway this is better now. Thanks